cdap-data-fabric-tests/src/test/java/co/cask/cdap/data/stream/StreamFileSizeManagerTest.java

```diff
@@ -0,0 +1,219 @@
+import co.cask.cdap.common.metrics.MetricsCollectionService;
+import co.cask.cdap.common.metrics.NoOpMetricsCollectionService;
+import co.cask.cdap.data.runtime.DataFabricModules;
+import co.cask.cdap.data.runtime.DataSetServiceModules;
+import co.cask.cdap.data.runtime.DataSetsModules;
+import co.cask.cdap.data.runtime.LocationStreamFileWriterFactory;
+import co.cask.cdap.data.stream.service.StreamHttpService;
+import co.cask.cdap.data.stream.service.StreamServiceRuntimeModule;
+import co.cask.cdap.data.stream.service.heartbeat.HeartbeatPublisher;
+import co.cask.cdap.data.stream.service.heartbeat.StreamWriterHeartbeat;
+import co.cask.cdap.data2.datafabric.dataset.service.DatasetService;
+import co.cask.cdap.data2.datafabric.dataset.service.executor.DatasetOpExecutor;
+import co.cask.cdap.data2.transaction.stream.StreamAdmin;
+import co.cask.cdap.data2.transaction.stream.StreamConsumerFactory;
+import co.cask.cdap.data2.transaction.stream.StreamConsumerStateStoreFactory;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamConsumerStateStoreFactory;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileAdmin;
+import co.cask.cdap.data2.transaction.stream.leveldb.LevelDBStreamFileConsumerFactory;
+import co.cask.cdap.explore.guice.ExploreClientModule;
+import co.cask.cdap.gateway.auth.AuthModule;
+import co.cask.tephra.TransactionManager;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.util.Modules;
+import org.apache.hadoop.conf.Configuration;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public class StreamFileSizeManagerTest {
+  private static final String API_KEY = "SampleTestApiKey";
+  private static final byte[] TWO_BYTES = new byte[] { 'a', 'b' };
+
+  private static final String HOSTNAME = "127.0.0.1";
+  private static int port;
+  private static CConfiguration conf;
+
+  private static Injector injector;
+  private static StreamHttpService streamHttpService;
+  private static TransactionManager txService;
+  private static DatasetOpExecutor dsOpService;
+  private static DatasetService datasetService;
+  private static MockHeartbeatPublisher heartbeatPublisher;
+  private static TemporaryFolder tmpFolder = new TemporaryFolder();
```

Comment:
If you annotate `tmpFolder` with `@Rule` then it will automatically get created on test start and deleted on test shutdown.